src/haxelib/client/Main.hx

```diff
@@ -843,6 +846,43 @@ class Main {
 		Cli.print('Local repository deleted ($path)');
 	}
 
+	/** Outputs a lock file with all the installed libraries **/
+	function lock() {
+		if (RepoManager.getLocalPath(Sys.getCwd()) == null) {
+			Cli.printWarning("no local repository found, dumping lock from global");
+		}
+
+		final scope = getScope();
+		final libraryInfo = scope.getArrayOfLibraryInfo();
+
+		// sort projects alphabetically
+		libraryInfo.sort(function(a, b) return Reflect.compare(a.name.toLowerCase(), b.name.toLowerCase()));
+
+		for (library in libraryInfo) {
+			if (library.devPath != null) {
+				Cli.printError('Error: version of "${library.name}" is "dev". The lockfile may not work properly in other environments!');
```

Comment:
If this is an error, does it mean I cannot use the command at all as soon as I have a single dev library? That seems rather limiting... Shouldn't it rather warn about this and ignore dev libraries?

Reply:
Ah, right I wanted to use "Warning" wording there.

I do want only warnings because I often have lock files that I want to keep working, but while actually working locally I also often work on some of these libs, so I use `haxelib dev` on them.

Current behavior warns about dev libs being present and so their line in output being potentially outdated, but still goes on.